src/tests/linalg/test_linalg.f90

```diff
@@ -424,7 +440,129 @@ subroutine test_trace_int64
     call check(trace(h) == sum(c(0:nd:2)), &
           msg="trace(h) == sum(c(0:nd:2)) failed.",warn=warn)
 
-  end subroutine
+  end subroutine test_trace_int64
+
+
+  subroutine test_outer_product_rsp
+    integer, parameter :: n = 2
+    real(sp) :: u(n), v(n), expected(n,n), diff(n,n)
+    write(*,*) "test_outer_product_rsp" 
+    u = [1.,2.]
+    v = [1.,3.]
+    expected = reshape([1.,2.,3.,6.],[n,n])
+    diff = expected - outer_product(u,v)
+    call check(all(abs(diff) < sptol), &
+         msg="all(abs(diff) < sptol) failed.",warn=warn)
```

Comment:
several tests have the same name, which can be tricky if an issue appear. I suggest to at least add the name of the function and the kind/type, e.g., 
```suggestion
         msg="outer rsp all(abs(diff) < sptol) failed.",warn=warn)
```

Reply:
I believe this is the only outstanding comment/change left.

I'm confused what the point of the `write (*,*), ...` lines are in all of the tests. These contain the test name, so perhaps we wouldn't need to put it in the `msg`? However, I've purposefully failed the `linalg` test and `grep`ed for the contents of the `write (*,*)` statement text and the `msg` text. I'm sure I've missed something, but to me it seems inconclusive whether there is enough info for a failed test to be traced back to the function.

Again, I tried to follow the previous examples in `test_linalg.f90`, which did not contain names in the `msg`s, but I'm not exactly sure how the testing and failure logging is done.

Reply:
If I am not wrong, the `msg` of the other tests are uniq. Therefore, it is easy to retrieve which tests failed. In your cases, the same `msg` is used by multiple tests, and it would need a bit more thinking to retrieve which test failed (by combining with the `write(*,*)`). I think it would be easier to trace a failed test if the `msg` and the `write(*,*)` would be combined into `msg`. However, I might be wrong.

Reply:
Yes, you are correct, they are all technically unique. The `msg`s don't necessarily have the function name in them (what I was expecting and looking for) but are unique enough to be `grep`ed, which I suppose is good enough for a quick traceback.

I'll make something like your suggested change for each test. I'm still not exactly clear on the point of `write` and `msg`, since I couldn't find either text even after a failed test, but this change will put `outer_product` in line with the others...

Anyway, thanks for catching this.

Reply:
I dug back into `check` and looked at `stdlib_error.f90` and still can't quite figure out what's happening with logs and error handling when testing. I'd like to better understand the error handling and logging for tests before I make this change.

From what I can deduce, this is the chain:
- start "execution" of `test_linalg.f90`
- `write(*,*), test_function_typeprecision` to terminal (or default out, not sure where that would be set?), though I'm not seeing this text in terminal when testing
- test fails in `check` statement
- since `warn=.false.` the program should stop via a call to `error_stop` (whose function definition I cannot find) with `msg`, which passes `msg` to `error stop` stopping execution

I am not sure if the above is quite right, but  the important source of confusion for me is: if a test fails, what is the proper way to determine this and trace back to the offending function? All I can see when testing is `stdlib_linalg........**Failed` and then
```
The following tests FAILED:
	 12 - linalg (Failed)
Errors while running CTest
make: *** [Makefile:130: test] Error 8
```
I don't even see info about a log file when running `cmake --build build --target test`, and even after `grep`ing for the `msg` (which I only know because I wrote the tests) I still turn up nothing related to why this test failed (note: I'm artificially making it fail). Perhaps I'm missing something, but with the provided info in terminal, it seems like it would be quite difficult to accurately trace this error, regardless of the `msg` text.

@ivan-pi Would you mind weighing in on this message and this comment thread?

Reply:
I am not sure what you mean, but you may want to look in `Testing/Temporary/LastTest.log`

Reply:
Crazy, not sure why that didn't come up via `grep`, but thanks.

Anyway, here's what I see when I purposefully fail `test_outer_product_rsp`:
```
...
 test_trace_int32
 test_trace_int64
 test_outer_product_rsp
 all(abs(diff) < sptol) failed.
ERROR STOP 1
```

The function name prints right before the `msg`, then execution of that test collection stops. Since the function name is right there, it should be just as easy to trace as any of the other tests.

However, one change I think might make sense is to pad `msg` with spaces on the left, so the failure message looks like it belongs to the failing test above it (for example):
```
...
 test_trace_int32
 test_trace_int64
 test_outer_product_rsp
    all(abs(diff) < sptol) failed.
ERROR STOP 1
```
but I think the log is clear enough as it stands, and this would probably be another PR itself.

Reply:
I always prefer to have the whole information in one line. But it could be matter of taste. If it is ok for others like that, it will be also ok for me